First Systems Engineering Best Practices sections

TheLastCicada opened the pull request:
3 sections added under the "Systems" heading: Nginx, PHP-FPM, and MySQL.  More intended to be added in the future.  

What needs help is:

* Banner for the Systems page
* Table of Contents formatting

Everything else looks to be working.  Proof-reading encouraged.

TheLastCicada committed:
Add Nginx section - testing

TheLastCicada committed:
Nginx content

TheLastCicada committed:
plumbing for the Nginx section

TheLastCicada committed:
Change "Nginx" to "Systems" in Nav; add text for MySQL and PHP-FPM

TheLastCicada committed:
Make the markdown look good

TheLastCicada committed:
Start updating the table of contents

TheLastCicada committed:
Text adjustments based on proofreading.  XML-RPC blocking.

TheLastCicada committed:
upload_max_file_size - additional notes regarding Nginx and CDNs

TheLastCicada committed:
Markdown code formatting improvement

TheLastCicada committed:
More markdown tweaks for bolding

tlovett1 commented on _includes/markdown/Systems.md:
```diff
@@ -0,0 +1,252 @@
+<h2 id="nginx" class="anchor-heading">Nginx{% include Util/top %}</h2>
+
+[Nginx](http://nginx.org/ "Nginx Web Server") is our preferred web server software at 10up. It has proven extremely stable, performant, and reliable at high scale and offers a powerful set of tools. This is not to imply there is anything wrong with using [Apache](https://httpd.apache.org/ "Apache Web Server") - we’ve worked on many high scale deployments that use Apache and mod_php that perform v ...
```
Link doesn't work. Why do we prefer nginx over Apache?

tlovett1 replied:
Nvm link works

TheLastCicada replied:
https://github.com/10up/Engineering-Best-Practices/pull/237/commits/c1423755a784b8b2525a49d957fff568818c740b adds some text addressing why Nginx over Apache.  It is intentionally weak in the language because Apache can be configured to work as well as Nginx, we just prefer the way Nginx does it.

tlovett1 commented on _includes/markdown/Systems.md:
An example file would be nice.

TheLastCicada replied:
An example php.ini file?

tlovett1 reviewed (commented):
Overall, this is awesome info.

One thing I noticed was capitalization inconsistencies throughout e.g. Memcached vs. memcached. github vs. GitHub, CentOS vs. centos, etc. Obviously not a show stopper but would be nice to address.

TheLastCicada committed:
Styling for Systems Engineering section and header image (props @tlovett1)

TheLastCicada committed:
Nginx vs Apache; capitalization consistency.

TheLastCicada commented:
https://github.com/10up/Engineering-Best-Practices/pull/237/commits/c1423755a784b8b2525a49d957fff568818c740b addresses the capitalization inconsistencies, but I'm nearly certain there are more of these issues in the text that we just haven't caught yet.  I expect those tweaks to be an on-going effort.  That being said, the most likely tech words that you pointed out (and a few others) have been addressed.

TheLastCicada committed:
Fix table of contents links for Systems section

ivankristianto commented on _includes/markdown/Systems.md:
```diff
@@ -0,0 +1,252 @@
+<h2 id="nginx" class="anchor-heading">Nginx{% include Util/top %}</h2>
+
+[Nginx](http://nginx.org/ "Nginx Web Server") is our preferred web server software at 10up. It has proven extremely stable, performant, and reliable at high scale and offers a powerful set of tools. This is not to imply there is anything wrong with using [Apache](https://httpd.apache.org/ "Apache Web Server") - we’ve worked on many high scale deployments that use Apache and mod_php that perform v ...
```
Found a typo here, `lightweigh`

ivankristianto commented:
I recommend to add extra information to have HTTP/2 enabled in Nginx Performance section

TheLastCicada committed:
Add performance recommendation for HTTP/2 and fix "lightweight" typo

TheLastCicada commented:
@ivankristianto I've added a section about HTTP/2 in the Performance part for Nginx and fixed the typo.  

I expect to add more content to these sections as we evolve our best practices and find additional improvements.  I also expect community members to have lots of good suggestions.  I do want to continue fixing typos and adding content to this, but feel like we are at a good place to merge where we can make tweaks after this is in the official document.
